sim/team-validator.ts

```diff
@@ -2571,6 +2580,12 @@ export class TeamValidator {
 							moveSources.pomegEggMoves = [move.id];
 						} else if (learnedGen < 6) {
 							limitedEggMove = move.id;
+						} else {
+							let motherLearnset;
+							if (species.mother) motherLearnset = this.dex.species.getLearnset(toID(species.mother));
+							if (motherLearnset && !motherLearnset[move.id]) {
```

Comment:
```suggestion
							if (!motherLearnset || !motherLearnset[move.id]) {
```
This should probably match line 2529. (Although, are we allowed optional chaining? If we are, this could become `if (!this.dex.species.getLearnset(toID(species.mother))?.[move.id]) {`, if that's not too much of a mouthful.)

Reply:
Pretty sure this isn't a necessary change. Male-only Pokemon with a possible mother should get their egg moves checked if the mother can't pass down the move, and for other male-only Pokemon, it's going to be a useless/incorrect check because the only fathers that can pass down the move are its evolutions, and any illegalities are handled by not including the move in the learnset.